frontend/class-twitter.php

```diff
@@ -460,7 +460,10 @@ private function image_thumbnail_output() {
 	 * @return bool
 	 */
 	private function image_from_content_output() {
-		if ( preg_match_all( '`<img [^>]+>`', $GLOBALS['post']->post_content, $matches ) ) {
+		global $post;
+		$content = apply_filters( 'wpseo_pre_analysis_post_content', $post->post_content, $post );
```

Comment:
Was following the same style as [get_singular_image() in WPSEO_OpenGraph](https://github.com/Yoast/wordpress-seo/blob/trunk/frontend/class-opengraph.php#L737), it uses ```global $post```.  I'll add the filter documentation.